packages/portalnetwork/src/networks/contentLookup.ts

```diff
@@ -65,121 +66,120 @@ export class ContentLookup {
       const dist = distance(peer.nodeId, this.contentId)
       this.lookupPeers.push({ nodeId: peer.nodeId, distance: dist })
     }
-    let finished = false
-    while (!finished) {
-      if (this.lookupPeers.length === 0) {
-        finished = true
-        this.network.metrics?.failedContentLookups.inc()
-        this.logger(
-          `No more peers to query.  Failed to retrieve ${toHexString(this.contentKey)} from network`,
-        )
-        return
+    while (!this.finished && this.lookupPeers.length > 0) {
+      // Process multiple peers in parallel
+      const peerBatch = this.lookupPeers.splice(0, 5)
+      const promises = peerBatch.map((peer) => this.processPeer(peer))
+
+      const results = await Promise.all(promises)
```

Comment:
I'm not sure I like the idea of `Promise.all` since it will reject if any of the batch of promises rejects (and it looks like we currently reject if the uTP stream times out without getting us the desired content) and I think that will end up crashing the lookup (unless we handle it somehow).

I've had a slight rethink of this and now think something like this might work better (not sure yet so would need to test out)
My rough idea is something like this:`
`while !finished {`
1) Add a new array of `this.lookupPromises`
2) `for (peer of firstFivePeers) void processPeer` 
3) Do `await new Promise((resolve) => setTimeout(3000, () => resolve))` or some suitable timeout
4) See if any of the promises have resolved.  If so, `for (peer of upToFiveNewPeers) void processPeer`
`}`

We'd need to pass a `resolve` function or something to `processPeer` that could call to mark "finished" and then close the lookup.  We also need to make sure that none of the promises reject without exceptions being caught. I think you can get away with `void promise.catch(() => handle exception)`.  Lemme do some experimenting over the next few days and see.

Reply:
Good point.  How about `Promise.allSettled`?  It resolves when all internal promises are resolves or rejected.